std/meta/index.zig

```diff
@@ -0,0 +1,691 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const debug = std.debug;
+const mem = std.mem;
+const math = std.math;
+
+pub const trait = @import("trait.zig");
+
+const TypeId = builtin.TypeId;
+const TypeInfo = builtin.TypeInfo;
+
+///Given a container or a pointer to a container, returns the container type.
+/// Otherwise it returns the type as-is. 
+pub fn UnwrapContainer(comptime T: type) type
+{
+    comptime
```

Comment:
Pretty sure functions returning `type` are always evaluated at comptime

Reply:
Chalk it up to paranoia from seeing too many bugs due to something being evaluated at runtime for non-obvious reasons.